packages/grafana-data/src/transformations/matchers/valueMatchers/substringMatchers.ts

```diff
@@ -0,0 +1,41 @@
+import { Field, FieldType } from '../../../types/dataFrame';
+import { ValueMatcherInfo } from '../../../types/transformations';
+import { ValueMatcherID } from '../ids';
+
+import { BasicValueMatcherOptions } from './types';
+
+const isSubstringMatcher: ValueMatcherInfo<BasicValueMatcherOptions> = {
+  id: ValueMatcherID.substring,
+  name: 'Contains Substring',
+  description: 'Match where value for given field is a substring to options value.',
+  get: (options) => {
+    return (valueIndex: number, field: Field) => {
+      const value = field.values[valueIndex];
+      return (value && value.includes(options.value)) || options.value === '';
+    };
+  },
+  getOptionsDisplayText: () => {
+    return `Matches all rows where field is similar to the value.`;
+  },
+  isApplicable: (field) => field.type === FieldType.string,
+  getDefaultOptions: () => ({ value: '' }),
+};
+
+const isNotSubstringValueMatcher: ValueMatcherInfo<BasicValueMatcherOptions> = {
+  id: ValueMatcherID.notSubstring,
+  name: 'Does not contain substring',
+  description: 'Match where value for given field is not a substring to options value.',
+  get: (options) => {
+    return (valueIndex: number, field: Field) => {
+      const value = field.values[valueIndex];
+      return options.value !== '' && (!value || !value.includes(options.value));
```

Comment:
`!value` wouldn't work for `'0'`, right?

Reply:
Fixed in https://github.com/grafana/grafana/pull/83548/commits/2f93ba25f71589e08539d4d1f3b54ef7fae684dc